c/datasetjson.c

```diff
@@ -1602,34 +1655,520 @@ void respondWithDataset(HttpResponse* response, char* absolutePath, int jsonMode
   }
 
 }
 
+/* new function for ENQ */
+void respondWithEnqueue(HttpResponse* response, char* absolutePath, int jsonMode) {
+
+  zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+      "ZSS1689I Begin %s\n", __FUNCTION__); 
+      pid_t pid = getpid();
+      zowelog(NULL, LOG_COMP_HTTPSERVER, ZOWE_LOG_INFO,
+        "ZSS1692I %s PID is %d\n", __FUNCTION__, pid);
+
+  HttpRequest *request = response->request;
+
+  if (!isDatasetPathValid(absolutePath)) {
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Invalid dataset name");
+    return;
+  }
+
+  DatasetName dsn;
+  DatasetMemberName memberName;
+  extractDatasetAndMemberName(absolutePath, &dsn, &memberName);
+  
+  DynallocDatasetName daDsn;
+  DynallocMemberName daMember;
+  memcpy(daDsn.name, dsn.value, sizeof(daDsn.name));
+  memcpy(daMember.name, memberName.value, sizeof(daMember.name));
+
+  /* new code to detect ENQ header begins */
+  // zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //           "ZSS0001I %s Requested dsn=\'%44.44s\', member=\'%8.8s\'\n", __FUNCTION__,
+  //           daDsn.name, daMember.name);
+
+  // int exclusiveENQ = 0; /* FALSE */
+  // /* obtain header name&value */
+  
+  // HttpHeader *enqHeader = getHeader(request, "ENQ");
+
+  // if (enqHeader == NULL){
+  //   zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //           "ZSS0002I %s HTTP request has no ENQ header\n", __FUNCTION__);
+  //   }
+  //   else {
+  //     zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //       "ZSS0003I %s HTTP request has ENQ header\n", __FUNCTION__);
+  //         zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //           "ZSS1621I %s Raw ENQ header value is %4.4s\n", __FUNCTION__, enqHeader->value );
+          
+  //         /* header values are in ASCII, so convert the local EBCDIC string before comparing them. */
+  //         char enqValue [5] = "true";
+  //         if(e2a(enqValue, 4) == -1){
+  //           zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //             "ZSS1626I %s Failed to convert true to ASCII\n", __FUNCTION__);
+  //         }else{            
+  //           if (!strncmp(enqHeader->value, enqValue, strlen(enqHeader->value)))
+  //             {
+  //               zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //                 "ZSS0005I %s HTTP request is asking for ENQ=true\n", __FUNCTION__);
+  //             /* request is asking for ENQ */
+  //               exclusiveENQ = 1; /* TRUE */
+  //             }
+  //             else {
+  //               zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO,
+  //                 "ZSS0006I %s HTTP request is NOT asking for ENQ=true\n", __FUNCTION__);
+  //               exclusiveENQ = 0;
+  //             }
+  //           }
+
+  //   } /* end ELSE request has header */
+  
+  /* new code ends   */
+
+  // DynallocDDName daDDname = {.name = "????????"};
+
+  // int daRC = RC_DYNALLOC_OK, daSysRC = 0, daSysRSN = 0;
```

Comment:
I believe you're going to remove all this commented out code eventually. Is that correct?

Reply:
Yes, that's part of code cleanup before MERGE.